### ISSUE TITLE
Use in highly-templated projects?

### ISSUE THREAD

**Issue by booleanbetrayal** [addyosmani/grunt-uncss#84]
I've got an Angular + Rails project which is extremely dynamic in nature, and as such, the best guarantee in CSS reduction would be to parse all available templates against my post-processed CSS amalgamation. Here's my (mildly sanitized) attempt at getting this working:

``` js
uncss: { 
    dist: {
        src: [
            'angular-app/**/views/**/*.html',
            'rails-app/views/**/*.html.erb'
        ],
        dest: 'dist/styles/main-tidied.css',
        options: {
            csspath: 'dist/styles/', // always being used as relative path?
            stylesheets: ['main.css']
        }
    }
},
```

This appears not to work, as each HTML template attempts to use its path for determining the target CSS path, during processing. As a result, I end up with errors like the following:

``` Shell
    Running "uncss:dist" (uncss) task
    Fatal error: UnCSS: could not open /Users/booleanbetrayal/some_project/angular-app/some_component/views/dist/styles/main.css
```

Presumably, being able to evaluate cssPath as an absolute path would resolve the above issue ... but in general, the tool would have an incomplete DOM hierarchy to evaluate style nesting, as the templates would be evaluated without parent context. CSS patterns would have to be fairly fuzzy to catch possible nesting scenarios.

So, in short, can Grunt + unCSS currently be used to address this scenario, or are there other alternatives that might be viable? Would love to systematically clobber CSS bloat with this  tool or something similar.

**Comment by XhmikosR** [comment]
This should be an upstream request, we support whatever uncss does.
